camera-manager.js

```diff
@@ -121,29 +121,96 @@ const cameraManager = {
       return value1 + (value2 - value1) * amount;
     }
 
+    function checkRayIntersection(origin,dirQuaternion,maxDistance)
+    {
+      let res = false;
+
+      let collision = physicsManager.raycast(origin, dirQuaternion);
+      if (collision) {
+        if (collision.distance <= maxDistance) {
+          res = true;
+        }
+      }
+
+      return res;
+    }
+
+    // Two rays from players position with small offset left and right
+    function checkSmallObjectIntersections(maxDistance)
+    {
+      let intersections = 0;
+
+      const localPlayer = metaversefile.useLocalPlayer();
+
+      let checkPoints = [];
```

Comment:
This is also an allocation which I don't think is necessary, especially if we collapse the raycast into a single call.